server/src/main/java/org/apache/iotdb/db/metadata/mtree/store/CachedMTreeStore.java

```diff
@@ -609,4 +621,19 @@ public void close() {
       }
     }
   }
+
+  private class CachedTraverserIterator extends AbstractTraverserIterator {
+    CachedTraverserIterator(IMTreeStore store, IMNode parent, Map<Integer, Template> templateMap)
+        throws MetadataException {
+      super(store, parent, templateMap);
+    }
+
+    @Override
+    public void close() {
+      if (nextMatchedNode != null && usingDirectChildrenIterator) {
+        unPin(nextMatchedNode);
+      }
+      super.close();
+    }
```

Comment:
Seems not strongly related to MTreeStore and invokes no private method of MTreeStore. Use an independent factory.

Reply:
done